collect_app/src/androidTest/java/org/odk/collect/android/feature/formentry/IntentGroupTest.java

```diff
@@ -80,17 +92,67 @@ public void appMissingErrorText_ShouldComeFromSpecialFormText() {
     }
 
     @Test
-    public void externalApp_ShouldPopulateFields() {
+    public void externalApp_ShouldPopulateFields() throws IOException {
+        // Check ImageWidget without answer
+        onView(withTagValue(is("ImageView"))).check(doesNotExist());
+        onView(withId(R.id.capture_image)).check(doesNotExist());
+        onView(withId(R.id.choose_image)).check(doesNotExist());
+
+        // Check AudioWidget without answer
+        onView(withId(R.id.audio_controller)).check(matches(not(isDisplayed())));
+
+        // Check VideoWidget without answer
+        onView(withId(R.id.play_video)).check(matches(isDisplayed()));
+        onView(withId(R.id.play_video)).check(matches(not(isEnabled())));
+
+        // Check ArbitraryFileWidget without answer
+        onView(withTagValue(is("ArbitraryFileWidgetAnswer"))).check(matches(not(isDisplayed())));
+
         Intent resultIntent = new Intent();
-        resultIntent.putExtra("sometext", "Here is a text value");
 
-        int randomInteger = (new Random()).nextInt(255);
-        resultIntent.putExtra("someinteger", randomInteger);
+        resultIntent.putExtra("questionInteger", "25");
+        resultIntent.putExtra("questionDecimal", "46.74");
+        resultIntent.putExtra("questionText", "sampleAnswer");
 
+        ClipData clipData = ClipData.newRawUri(null, null);
+        clipData.addItem(new ClipData.Item("questionImage", null, createTempFile("famous", "jpg")));
```

Comment:
According to our spec, an external app is supposed to:
> Add content URIs for the files to attach to the form to the return intent’s ClipData

> Add an intent extra with desired form field name and set its value to the content URI for the corresponding file.

Reply:
This test needs to be updated!

Reply:
Done.